File: Python/Problem339A_HelpfulMaths.py
def main():
    expression = input()
    length = len(expression)
    num1 = num2 = num3 = 0
    
    while length > 0:
        if expression[length-1] == '1':
            num1 += 1
        elif expression[length-1] == '2':
            num2 += 1
        else:
            num3 += 1
        length -= 2
    result = num1 * "1+" + num2 * "2+" + (num3) * "3+"
    print(result[:-1])

if __name__ == "__main__":
    main()

File: C++/Problem339A_HelpfulMaths.cpp
#include <iostream>
#include <cstring>
using namespace std;

int main(){
    string expression, result;
    cin >> expression;
    int length = expression.length();

    for(int i = 0; i  < length; i += 2){
        if(expression[i] == '1')
            result.append("1+");
    }
    
    for(int i = 0; i  < length; i += 2){
        if(expression[i] == '2')
            result.append("2+");
    }

    for(int i = 0; i  < length; i += 2){
        if(expression[i] == '3')
            result.append("3+");
    }

    result.pop_back();
    cout << result;


    return 0;
}